src/ALE/Recon1d_PPM_CWK.F90

```diff
@@ -215,6 +218,48 @@ real function dfdx(this, k, x)
 
 end function dfdx
 
+!> Solver for x: f(x)=t
+real function x(this, k, t)
+  class(PPM_CWK), intent(in) :: this !< This reconstruction
+  integer,        intent(in) :: k    !< Cell number
+  real,           intent(in) :: t    !< Value to solve for [A]
+  real :: xl, xr, xo ! Left/right bounds and guess [nondim]
+  real :: fl, fr ! Left right values [A]
+  real :: slp ! Difference across cell or derivative wrt nondim x [A]
+  real :: f_at_x ! Value at current x [A]
+  integer :: iter
+
+  slp = this%ur(k) - this%ul(k)
+  if ( abs(slp) > 0. ) then
+    xl = 0.
+    xr = 1.
+    xo = ( t - this%ul(k) ) / slp ! First guess by regula falsi
+    f_at_x = this%f(k, xo)
+    do iter = 1,5
```

Comment:
Please add a comment explaining why this code is using Newton's method to solve for the root of a quadratic equation rather than just using the closed-form solution.  (Could it be that the profile is quadratic in T and S, but with a nonlinear equation of state the function might be nonlinear for density, for example?)